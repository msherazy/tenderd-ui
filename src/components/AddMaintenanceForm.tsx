import React from 'react';
import type { MaintenanceFormData } from '../types';
import { FormInput } from './FormComponents';

interface AddMaintenanceFormProps {
	formData: MaintenanceFormData;
	formErrors: Record<string, string>;
	onFormChange: (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
	onFormSubmit: (e: React.FormEvent) => void;
	onCancel: () => void;
}

export const AddMaintenanceForm: React.FC<AddMaintenanceFormProps> = ({
	formData,
	formErrors,
	onFormChange,
	onFormSubmit,
	onCancel,
}) => {
	return (
		<div className="bg-white dark:bg-gray-800 shadow-lg rounded-lg p-6 border border-gray-100 dark:border-gray-700">
			<form onSubmit={onFormSubmit} className="dark:bg-gray-800">
				<div className="grid grid-cols-1 md:grid-cols-2 gap-6">
					{/* Maintenance Details Section */}
					<div className="space-y-4 bg-gray-50 dark:bg-gray-800 p-5 rounded-lg shadow-sm">
						<h3 className="text-lg font-medium text-gray-900 dark:text-gray-100 mb-4 border-b border-gray-200 dark:border-gray-700 pb-2">
							Maintenance Details
						</h3>
						<FormInput
							label="Date"
							name="date"
							type="date"
							value={formData.date}
							onChange={onFormChange}
							error={formErrors.date}
							required
						/>
						<FormInput
							label="Description"
							name="description"
							type="text"
							value={formData.description}
							onChange={onFormChange}
							error={formErrors.description}
							required
						/>
						<FormInput
							label="Cost"
							name="cost"
							type="number"
							value={formData.cost.toString()}
							onChange={onFormChange}
							error={formErrors.cost}
							min="0"
							required
						/>
					</div>

					{/* Additional Information Section */}
					<div className="space-y-4 bg-gray-50 dark:bg-gray-800 p-5 rounded-lg shadow-sm">
						<h3 className="text-lg font-medium text-gray-900 dark:text-gray-100 mb-4 border-b border-gray-200 dark:border-gray-700 pb-2">
							Additional Information
						</h3>
						<FormInput
							label="Mileage"
							name="mileage"
							type="number"
							value={formData.mileage?.toString() || ''}
							onChange={onFormChange}
							error={formErrors.mileage}
							min="0"
						/>
						<FormInput
							label="Service Center"
							name="serviceCenter"
							type="text"
							value={formData.serviceCenter}
							onChange={onFormChange}
							error={formErrors.serviceCenter}
						/>
						<FormInput
							label="Notes"
							name="notes"
							type="text"
							value={formData.notes}
							onChange={onFormChange}
							error={formErrors.notes}
						/>
						<FormInput
							label="Next Due Date"
							name="nextDueDate"
							type="date"
							value={formData.nextDueDate || ''}
							onChange={onFormChange}
							error={formErrors.nextDueDate}
						/>
					</div>
				</div>

				<div className="mt-8">
					<button
						type="submit"
						className="w-full md:w-auto px-6 py-3 bg-indigo-600 dark:bg-indigo-500 text-white font-medium rounded-md hover:bg-indigo-700 dark:hover:bg-indigo-600 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 dark:focus:ring-offset-gray-800"
					>
						Add Maintenance
					</button>
				</div>
			</form>
		</div>
	);
};
