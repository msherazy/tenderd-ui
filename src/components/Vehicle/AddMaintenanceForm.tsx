import React, { useState, useEffect } from 'react';
import type { MaintenanceFormData } from '../../types';
import { FormInput } from '../Form';
import { Index } from '../Button';
import { t } from '../../utils/locale';

interface AddMaintenanceFormProps {
	formData: MaintenanceFormData;
	formErrors: Record<string, string>;
	onFormChange: (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
	onFormSubmit: (e: React.FormEvent) => void;
	onCancel: () => void;
}

export const AddMaintenanceForm: React.FC<AddMaintenanceFormProps> = ({
	formData,
	formErrors: initialFormErrors,
	onFormChange,
	onFormSubmit,
}) => {
	// Local state for form errors so we can track them internally before submission
	const [formErrors, setFormErrors] = useState<Record<string, string>>(initialFormErrors);

	// Function to update parent's form errors and  local errors
	const updateFormErrors = (field: string, message: string) => {
		setFormErrors(prev => ({
			...prev,
			[field]: message,
		}));

		const simulatedEvent = {
			target: {
				name: field,
				value: formData[field as keyof MaintenanceFormData] || '',
			},
		} as React.ChangeEvent<HTMLInputElement>;
		onFormChange(simulatedEvent);
	};

	// Update local errors when initialFormErrors changes
	useEffect(() => {
		setFormErrors(initialFormErrors);
	}, [initialFormErrors]);

	const validateForm = (): boolean => {
		const errors: Record<string, string> = {};
		const today = new Date();

		// Required field validations
		if (!formData.date) {
			errors.date = 'Date is required';
		} else if (new Date(formData.date) > today) {
			errors.date = 'Date cannot be in the future';
		}

		if (!formData.description?.trim()) {
			errors.description = 'Description is required';
		}

		// Cost validation
		if (formData.cost === undefined || formData.cost === null) {
			errors.cost = 'Cost is required';
		} else if (formData.cost < 0) {
			errors.cost = 'Cost cannot be negative';
		} else if (formData.cost === 0) {
			errors.cost = 'Are you sure the cost is zero?';
		} else if (formData.cost > 10000) {
			errors.cost = 'Cost seems unusually high. Please verify.';
		}

		// Mileage validation
		if (formData.mileage === undefined || formData.mileage === null) {
			errors.mileage = 'Mileage is required';
		} else if (formData.mileage < 0) {
			errors.mileage = 'Mileage cannot be negative';
		} else if (formData.mileage > 1000000) {
			errors.mileage = 'Mileage seems unusually high. Please verify.';
		}

		// Next due date validation
		if (!formData.nextDueDate) {
			errors.nextDueDate = 'Next due date is required';
		} else {
			const nextDueDate = new Date(formData.nextDueDate);
			const maintenanceDate = formData.date ? new Date(formData.date) : today;

			if (nextDueDate <= maintenanceDate) {
				errors.nextDueDate = 'Next due date must be after the maintenance date';
			} else {
				const twoYearsFromNow = new Date();
				twoYearsFromNow.setFullYear(twoYearsFromNow.getFullYear() + 2);

				if (nextDueDate > twoYearsFromNow) {
					errors.nextDueDate = 'Next due date is very far in the future. Please verify.';
				}
			}
		}

		const hasErrors = Object.keys(errors).length > 0;
		if (hasErrors) {
			Object.entries(errors).forEach(([field, message]) => {
				updateFormErrors(field, message);
			});
		} else {
			Object.keys(formErrors).forEach(field => {
				updateFormErrors(field, '');
			});
		}

		return !hasErrors;
	};

	const handleSubmit = (e: React.FormEvent) => {
		e.preventDefault();
		if (validateForm()) {
			onFormSubmit(e);
		}
	};

	return (
		<div className="bg-white shadow-lg rounded-lg p-6 border border-gray-100 custom-card-style">
			<form onSubmit={handleSubmit} role="form">
				<div className="grid grid-cols-1 md:grid-cols-2 gap-6">
					{/* Maintenance Details Section */}
					<div className="space-y-4 bg-gray-50 p-5 rounded-lg shadow-sm">
						<h3 className="text-lg font-medium text-dark mb-4 border-b border-gray-200 pb-2">
							Maintenance Details
						</h3>
						<FormInput
							label="Date"
							name="date"
							type="date"
							value={formData.date}
							onChange={onFormChange}
							error={formErrors.date}
							required
							data-testid="maintenance-date-input"
						/>
						<FormInput
							label="Description"
							name="description"
							type="text"
							value={formData.description}
							onChange={onFormChange}
							error={formErrors.description}
							required
							data-testid="maintenance-description-input"
						/>
						<FormInput
							label="Cost"
							name="cost"
							type="number"
							value={formData.cost.toString()}
							onChange={onFormChange}
							error={formErrors.cost}
							min="0"
							required
							data-testid="maintenance-cost-input"
						/>
					</div>

					{/* Additional Information Section */}
					<div className="space-y-4 bg-gray-50 p-5 rounded-lg shadow-sm">
						<h3 className="text-lg font-medium mb-4 border-b border-gray-200 pb-2">
							Additional Information
						</h3>
						<FormInput
							label="Mileage"
							name="mileage"
							type="number"
							value={formData.mileage?.toString() || ''}
							onChange={onFormChange}
							error={formErrors.mileage}
							min="0"
							required
							data-testid="maintenance-mileage-input"
						/>
						<FormInput
							label="Service Center"
							name="serviceCenter"
							type="text"
							value={formData.serviceCenter}
							onChange={onFormChange}
							error={formErrors.serviceCenter}
							data-testid="maintenance-service-center-input"
						/>
						<FormInput
							label="Notes"
							name="notes"
							type="text"
							value={formData.notes}
							onChange={onFormChange}
							error={formErrors.notes}
							data-testid="maintenance-notes-input"
						/>
						<FormInput
							label="Next Due Date"
							name="nextDueDate"
							type="date"
							value={formData.nextDueDate || ''}
							onChange={onFormChange}
							error={formErrors.nextDueDate}
							required
							data-testid="maintenance-next-due-date-input"
						/>
					</div>
				</div>

				<div className="mt-8 flex justify-end space-x-3">
					<Index type="submit" variant="primary">
						{t('SUBMIT')}
					</Index>
				</div>
			</form>
		</div>
	);
};
