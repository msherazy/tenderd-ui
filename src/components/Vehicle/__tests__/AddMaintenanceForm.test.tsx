import { render, screen, fireEvent } from '@testing-library/react';
import { AddMaintenanceForm } from '../AddMaintenanceForm';

describe('AddMaintenanceForm', () => {
  const today = new Date();
  const nextMonth = new Date(today);
  nextMonth.setMonth(today.getMonth() + 1);

  const todayStr = today.toISOString().split('T')[0];
  const nextMonthStr = nextMonth.toISOString().split('T')[0];

  const defaultProps = {
    formData: {
      date: todayStr,
      description: 'Oil Change',
      cost: 50,
      mileage: 5000,
      serviceCenter: 'Test Center',
      notes: 'Test Notes',
      nextDueDate: nextMonthStr
    },
    formErrors: {},
    onFormChange: jest.fn(),
    onFormSubmit: jest.fn(),
    onCancel: jest.fn()
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders fields and submit button', () => {
    const { container } = render(<AddMaintenanceForm {...defaultProps} />);
    expect(screen.getByTestId('maintenance-date-input')).toBeInTheDocument();
    expect(screen.getByTestId('maintenance-description-input')).toBeInTheDocument();
    expect(screen.getByTestId('maintenance-cost-input')).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /submit/i })).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('calls onFormSubmit when form is submitted', async () => {
    const handleSubmit = jest.fn(e => e.preventDefault());
    const { container } = render(
      <AddMaintenanceForm
        {...defaultProps}
        onFormSubmit={handleSubmit}
      />
    );
    const form = screen.getByRole('form');
    fireEvent.submit(form);
    expect(handleSubmit).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });

  it('shows form errors when present', () => {
    const { container } = render(
      <AddMaintenanceForm
        {...defaultProps}
        formErrors={{
          date: 'Date is required',
          description: 'Description is required'
        }}
      />
    );

    expect(screen.getByText('Date is required')).toBeInTheDocument();
    expect(screen.getByText('Description is required')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
