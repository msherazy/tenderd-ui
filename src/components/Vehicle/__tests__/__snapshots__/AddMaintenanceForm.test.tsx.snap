// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMaintenanceForm calls onFormSubmit when form is submitted 1`] = `
<div>
  <div
    class="bg-white shadow-lg rounded-lg p-6 border border-gray-100 custom-card-style"
  >
    <form
      role="form"
    >
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-6"
      >
        <div
          class="space-y-4 bg-gray-50 p-5 rounded-lg shadow-sm"
        >
          <h3
            class="text-lg font-medium text-dark mb-4 border-b border-gray-200 pb-2"
          >
            Maintenance Details
          </h3>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-date"
            >
              Date
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-date-input"
              id="input-date"
              name="date"
              required=""
              type="date"
              value="2025-05-25"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-description"
            >
              Description
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-description-input"
              id="input-description"
              name="description"
              required=""
              type="text"
              value="Oil Change"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-cost"
            >
              Cost
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-cost-input"
              id="input-cost"
              min="0"
              name="cost"
              required=""
              type="number"
              value="50"
            />
          </div>
        </div>
        <div
          class="space-y-4 bg-gray-50 p-5 rounded-lg shadow-sm"
        >
          <h3
            class="text-lg font-medium mb-4 border-b border-gray-200 pb-2"
          >
            Additional Information
          </h3>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-mileage"
            >
              Mileage
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-mileage-input"
              id="input-mileage"
              min="0"
              name="mileage"
              required=""
              type="number"
              value="5000"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-serviceCenter"
            >
              Service Center
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-service-center-input"
              id="input-serviceCenter"
              name="serviceCenter"
              type="text"
              value="Test Center"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-notes"
            >
              Notes
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-notes-input"
              id="input-notes"
              name="notes"
              type="text"
              value="Test Notes"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-nextDueDate"
            >
              Next Due Date
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-next-due-date-input"
              id="input-nextDueDate"
              name="nextDueDate"
              required=""
              type="date"
              value="2025-06-25"
            />
          </div>
        </div>
      </div>
      <div
        class="mt-8 flex justify-end space-x-3"
      >
        <button
          class="inline-flex items-center justify-center px-4 py-2 rounded font-semibold focus:outline-none transition disabled:opacity-60 w-auto hover:opacity-90 focus:ring-2 focus:ring-offset-2 focus:ring-primary"
          style="color: rgb(255, 255, 255);"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`AddMaintenanceForm renders fields and submit button 1`] = `
<div>
  <div
    class="bg-white shadow-lg rounded-lg p-6 border border-gray-100 custom-card-style"
  >
    <form
      role="form"
    >
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-6"
      >
        <div
          class="space-y-4 bg-gray-50 p-5 rounded-lg shadow-sm"
        >
          <h3
            class="text-lg font-medium text-dark mb-4 border-b border-gray-200 pb-2"
          >
            Maintenance Details
          </h3>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-date"
            >
              Date
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-date-input"
              id="input-date"
              name="date"
              required=""
              type="date"
              value="2025-05-25"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-description"
            >
              Description
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-description-input"
              id="input-description"
              name="description"
              required=""
              type="text"
              value="Oil Change"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-cost"
            >
              Cost
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-cost-input"
              id="input-cost"
              min="0"
              name="cost"
              required=""
              type="number"
              value="50"
            />
          </div>
        </div>
        <div
          class="space-y-4 bg-gray-50 p-5 rounded-lg shadow-sm"
        >
          <h3
            class="text-lg font-medium mb-4 border-b border-gray-200 pb-2"
          >
            Additional Information
          </h3>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-mileage"
            >
              Mileage
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-mileage-input"
              id="input-mileage"
              min="0"
              name="mileage"
              required=""
              type="number"
              value="5000"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-serviceCenter"
            >
              Service Center
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-service-center-input"
              id="input-serviceCenter"
              name="serviceCenter"
              type="text"
              value="Test Center"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-notes"
            >
              Notes
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-notes-input"
              id="input-notes"
              name="notes"
              type="text"
              value="Test Notes"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-nextDueDate"
            >
              Next Due Date
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-next-due-date-input"
              id="input-nextDueDate"
              name="nextDueDate"
              required=""
              type="date"
              value="2025-06-25"
            />
          </div>
        </div>
      </div>
      <div
        class="mt-8 flex justify-end space-x-3"
      >
        <button
          class="inline-flex items-center justify-center px-4 py-2 rounded font-semibold focus:outline-none transition disabled:opacity-60 w-auto hover:opacity-90 focus:ring-2 focus:ring-offset-2 focus:ring-primary"
          style="color: rgb(255, 255, 255);"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`AddMaintenanceForm shows form errors when present 1`] = `
<div>
  <div
    class="bg-white shadow-lg rounded-lg p-6 border border-gray-100 custom-card-style"
  >
    <form
      role="form"
    >
      <div
        class="grid grid-cols-1 md:grid-cols-2 gap-6"
      >
        <div
          class="space-y-4 bg-gray-50 p-5 rounded-lg shadow-sm"
        >
          <h3
            class="text-lg font-medium text-dark mb-4 border-b border-gray-200 pb-2"
          >
            Maintenance Details
          </h3>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-date"
            >
              Date
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 
  border-red-300 focus:ring-red-500 focus:border-red-500
"
              data-testid="maintenance-date-input"
              id="input-date"
              name="date"
              required=""
              type="date"
              value="2025-05-25"
            />
            <p
              class="mt-1 text-sm text-red-600"
            >
              Date is required
            </p>
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-description"
            >
              Description
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 
  border-red-300 focus:ring-red-500 focus:border-red-500
"
              data-testid="maintenance-description-input"
              id="input-description"
              name="description"
              required=""
              type="text"
              value="Oil Change"
            />
            <p
              class="mt-1 text-sm text-red-600"
            >
              Description is required
            </p>
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-cost"
            >
              Cost
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-cost-input"
              id="input-cost"
              min="0"
              name="cost"
              required=""
              type="number"
              value="50"
            />
          </div>
        </div>
        <div
          class="space-y-4 bg-gray-50 p-5 rounded-lg shadow-sm"
        >
          <h3
            class="text-lg font-medium mb-4 border-b border-gray-200 pb-2"
          >
            Additional Information
          </h3>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-mileage"
            >
              Mileage
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-mileage-input"
              id="input-mileage"
              min="0"
              name="mileage"
              required=""
              type="number"
              value="5000"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-serviceCenter"
            >
              Service Center
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-service-center-input"
              id="input-serviceCenter"
              name="serviceCenter"
              type="text"
              value="Test Center"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-notes"
            >
              Notes
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-notes-input"
              id="input-notes"
              name="notes"
              type="text"
              value="Test Notes"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="
  block text-sm font-medium text-gray-700 mb-1
"
              for="input-nextDueDate"
            >
              Next Due Date
              <span
                class="text-red-600 ml-1"
              >
                *
              </span>
            </label>
            <input
              class="
  block w-full rounded-md shadow-sm px-3 py-2
  border-gray-300 focus:ring-indigo-500 focus:border-indigo-500
 "
              data-testid="maintenance-next-due-date-input"
              id="input-nextDueDate"
              name="nextDueDate"
              required=""
              type="date"
              value="2025-06-25"
            />
          </div>
        </div>
      </div>
      <div
        class="mt-8 flex justify-end space-x-3"
      >
        <button
          class="inline-flex items-center justify-center px-4 py-2 rounded font-semibold focus:outline-none transition disabled:opacity-60 w-auto hover:opacity-90 focus:ring-2 focus:ring-offset-2 focus:ring-primary"
          style="color: rgb(255, 255, 255);"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
